### ISSUE TITLE
IrrepsArray.transform_by_angles and LinearSHTP does not support batch operation

### ISSUE THREAD

**Issue by AdaaaWang** [e3nn/e3nn-jax#70]
the experimental class LinearSHTP does not support batch opertation, i.e., it only supports input of dimension [1, dim of Irreps]. However, eSCN works on IrrepsArrays of dimension [#nodes, dim of Irreps]. I wonder if this is because the rotation procedure in eSCN, implemented by transform_by_angles in LinearSHTP, only supports rotation by the same angle for all node features, while eSCN requires rotation by different angles for different nodes(i.e., output[i,...] = D[i,...] * input[i,...] for each node index i, where D contains representations of rotation matrices). 

to support such kind of operation, a quick solution that comes up to my mind is to rewrite transform_by_angles as

D = {
            ir: ir.D_from_angles(alpha, beta, gamma, k)
            for ir in {ir for _, ir in self.irreps}
        }
        if inverse:
            D = {ir: jnp.swapaxes(D[ir], -2, -1) for ir in D}
        new_chunks = [
                jnp.einsum("ij,...uj->...ui", D[ir], x)
                if x is not None
                else None
            )
            for (mul, ir), x in zip(self.irreps, self.chunks)
        ]
        return e3nn.from_chunks(self.irreps, new_chunks, self.shape[:-1], self.dtype)

where now alpha, beta, gammas are arrays of shape self.shape[:-1]. 

Since I am not a developer of e3nn, I may not be able to realize some subtleties regarding implementations. Are there any concerns about implementing this function?

(although I can also use vmap and the current LinearSHTP to implement eSCN, this may require initializing an nn.Module in __call__. I wonder if this will bring problems.)

**Comment by mariogeiger** [comment]
Yes I expect you to use vmap

**Comment by AdaaaWang** [comment]
Thanks!
